### PR TITLE
[TVMScript] Syntax sugar for reads & writes

### PR DESCRIPTION
**Pull request by shingjan**
This PR intends to add syntax sugar for `T.reads` and `T.writes`
Before this PR:
```
T.reads([C[vi, vj], A_shared[vk, vi], B[vk, vj]])
T.writes([C[vi, vj]])
```

After this PR:
```
T.reads(C[vi, vj], A_shared[vk, vi], B[vk, vj])
T.writes(C[vi, vj])
```

cc: @vinx13 @junrushao1994 @Hzfengsy

### CODE REVIEW

**Comment by vinx13** [comment]
please also update type stubs (adding an overload like `reads(*read_regions: BufferSlice`)

**Comment by shingjan** [comment]
@vinx13 stub is updated

**Comment by vinx13** [comment]
please fix CI error

**Comment by shingjan** [comment]
@vinx13 I fixed the CI error by removing the `test_mismatch_args` in `test_tvmscript_error_report` as the error condition of that test, `T.reads(A[...], A[...])` no longer holds after this PR. @Hzfengsy It would be great if you can take a look and confirm it. Thanks!

File: python/tvm/script/tir/__init__.pyi [hunk 1]
```diff
@@ -211,8 +211,12 @@ def alloc_buffer(
...
 """
 
-def reads(read_regions: Union[BufferSlice, List[BufferSlice]]) -> None: ...
-def writes(write_region: Union[BufferSlice, List[BufferSlice]]) -> None: ...
+def reads(
+    read_regions: Union[BufferSlice, List[BufferSlice]], *other_regions: BufferSlice
+) -> None: ...
+def writes(
+    write_region: Union[BufferSlice, List[BufferSlice]], *other_regions: BufferSlice
+) -> None: ...
```
Comment (junrushao1994) [inline python/tvm/script/tir/__init__.pyi:219]: Just curious, is it possible to use @overload to separate the logic more clearly?

**Comment by junrushao1994** [comment]
Let's also improve the printer to print the sugared syntax by default :-)

File: python/tvm/script/tir/__init__.pyi (hunk 1 above)
Comment (shingjan) [reply python/tvm/script/tir/__init__.pyi:219]: using @overload in TIR script will cause an error as @overload will call an anonymous function which is not allowed currently [here](https://github.com/apache/tvm/blob/3f1c931aa1414b69103c8c3ca019a9f55b5cb7a7/python/tvm/script/utils.py#L47) in TVM.

Comment (shingjan) [reply python/tvm/script/tir/__init__.pyi]: done

File: python/tvm/script/tir/__init__.pyi [hunk 2]
```diff
@@ -210,9 +210,14 @@ def alloc_buffer(
 """
 special_stmt - Reads/Writes
 """
-
+@overload
 def reads(read_regions: Union[BufferSlice, List[BufferSlice]]) -> None: ...
+@overload
+def reads(read_regions: BufferSlice, *other_regions: BufferSlice) -> None: ...
```
Comment (vinx13) [inline python/tvm/script/tir/__init__.pyi:216]: From the user-facing side, if we have two overloading versions I don't think `other_regions` is necessary. We can instead do something like
```
@overload
def reads(read_regions: List[BufferSlice]) -> None: ...
@overload
def reads(*read_regions: BufferSlice) -> None: ...
```

Comment (shingjan) [reply python/tvm/script/tir/__init__.pyi]: done

**Review by vinx13 — APPROVED** [review]

**Review by junrushao1994 — APPROVED** [review]
Okay. Let's move quickly to get this PR in, and improve the printer in a subsequent PR.

**Pull request merged**